src/Application/Matching/Adapter.php

```diff
@@ -46,29 +43,19 @@ class Adapter
 
     public function __construct(
         private RealTimeMatchingStorage $storage,
-        private EntityManagerInterface $entityManager,
         private LoggerInterface $logger,
     ) {
     }
 
     /**
-     * @param numeric-string $amount
-     */
-    public function addAmount(CampaignFunding $funding, string $amount): string
-    {
-        $fundBalance = $this->addAmountWithoutSavingFundingsToDB($funding, $amount);
-        $this->saveFundingsToDatabase();
-
-        return $fundBalance;
-    }
-
-
-    /**
+     * Callers are responsible for flushing after this works on `CampaignFunding`s – typically once per larger operation
+     * if looping over many.
+     *
      * @param CampaignFunding $funding
      * @param numeric-string $amount
      * @return numeric-string New fund balance
      */
-    private function addAmountWithoutSavingFundingsToDB(CampaignFunding $funding, string $amount): string
+    public function addAmount(CampaignFunding $funding, string $amount): string
```

Comment:
Could we also rename `subtractAmountWithoutSavingToDB` to just `subtractAmount` for symmetry?

Reply:
Yep! Done in c55d239727d83a90c1b0bd9b6472a07f482dc6ae